src/TextForEvent.js

```diff
@@ -134,6 +134,38 @@ function textForTombstoneEvent(ev) {
     return _t('%(senderDisplayName)s upgraded this room.', {senderDisplayName});
 }
 
+function textForJoinRulesEvent(ev) {
+    const senderDisplayName = ev.sender && ev.sender.name ? ev.sender.name : ev.getSender();
+    switch (ev.getContent().join_rule) {
+        case "public":
+            return _t('%(senderDisplayName)s made the room public to whoever knows the link.', {senderDisplayName});
+        case "invite":
+            return _t('%(senderDisplayName)s made the room invite only.', {senderDisplayName});
+        default:
+            // The spec supports "knock" and "private", however nothing implements these.
+            return _t('%(senderDisplayName)s changed the join rules to %(rule)s', {
```

Comment:
There appears to be only one join rule, so maybe this is more correct?

```suggestion
            return _t('%(senderDisplayName)s changed the join rule to %(rule)s', {
```

Reply:
Manually applied because i18n